examples/client-demo/src/main/java/com/fiskaly/kassensichv/demo/interceptors/ResponseTimeLoggingInterceptor.java

```diff
@@ -0,0 +1,96 @@
+package com.fiskaly.kassensichv.demo.interceptors;
+
+import okhttp3.Headers;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class ResponseTimeLoggingInterceptor implements Interceptor  {
+    private class LoggingRecord {
+        private String timestamp;
+        private String requestId;
+        private String method;
+        private String url;
+        private long clientDelta;
+        private long serverDelta;
+
+        public LoggingRecord(String timestamp, String requestId, String method,
+                             String url, long clientDelta, long serverDelta) {
+            this.timestamp = timestamp;
+            this.requestId = requestId;
+            this.method = method;
+            this.url = url;
+            this.clientDelta = clientDelta;
+            this.serverDelta = serverDelta;
+        }
+
+        @Override
+        public String toString() {
+            return timestamp + ";"
+                    + requestId + ";"
+                    + method + ";"
+                    + url + ";"
+                    + Long.toString(clientDelta) + ";"
+                    + Long.toString(serverDelta);
+        }
+    }
+
+    private OutputStream logStream;
+    private OutputStream errorStream;
+    private OutputStreamWriter writer;
+    private OutputStreamWriter errorWriter;
+
+    public ResponseTimeLoggingInterceptor(OutputStream logStream, OutputStream errorStream) {
+        this.logStream = logStream;
+        this.writer = new OutputStreamWriter(logStream, Charset.forName(StandardCharsets.UTF_8.name()));
+
+        this.errorStream = errorStream;
+        this.errorWriter = new OutputStreamWriter(errorStream, Charset.forName(StandardCharsets.UTF_8.name()));
+    }
+
+    @Override
+    public Response intercept(@NotNull Chain chain) throws IOException {
```

Comment:
Not sure if we should keep the NotNull Annotation in. @pgaubatz ?

Reply:
why shouldn't we?